Storage/src/Connection/Rest.php

```diff
@@ -431,7 +431,11 @@ private function resolveUploadOptions(array $args)
         }
 
         $args['metadata']['name'] = $args['name'];
+        if (isset($args['retention'])) {
+            $args['metadata']['retention'] = $args['retention'];
+        }
         unset($args['name']);
+        unset($args['retention']);
```

Comment:
nit, but why don't we move this into the if block above, as it flows more logically
```suggestion
        if (isset($args['retention'])) {
            $args['metadata']['retention'] = $args['retention'];
            unset($args['retention']);
        }
        unset($args['name']);
```